### PR TITLE
Task: Add Message Templates to SNS Notifier

### PR DESCRIPTION
**Pull request by niklasR**
* Template for Zenoss (for 247 monitoring)
* Plaintext template for SMS alerting

### CODE REVIEW

File: pkg/services/alerting/notifiers/aws_sns.go [hunk 1]
```diff
@@ -54,50 +84,109 @@ func NewAwsSnsNotifier(model *m.AlertNotification) (alerting.Notifier, error) {
 		return nil, alerting.ValidationError{Reason: "Could not find topic arn property in settings"}
 	}
 
+	messageTemplate := model.Settings.Get("message_template").MustString()
+	if messageTemplate == "" {
+		return nil, alerting.ValidationError{Reason: "Could not find Message Templase property in settings"}
```
Comment (zya) [inline pkg/services/alerting/notifiers/aws_sns.go]: Templase -> Template

**Comment by zya** [comment]
So, if I understand correctly, this PR formats the SNS notification the way opts expects it to be formatted? Previously, it was just a place holder?

**Comment by zya** [comment]
Also, can you explain why is there two different formats?

**Comment by niklasR** [comment]
> So, if I understand correctly, this PR formats the SNS notification the way opts expects it to be formatted? Previously, it was just a place holder?

Yes, before it was just a JSON blob with some important alert stuff, now it's in the correct format to be picked up by 247 (hopefully)

> Also, can you explain why is there two different formats?

The plaintext one is to send SMS alerts, so we can retire Twilio and our Webhook/Lambda

**Comment by zya** [comment]
So, the SMS alert one, is that using SNS too? Managed by us or ops?

**Comment by niklasR** [comment]
> So, the SMS alert one, is that using SNS too? Managed by us or ops?

Yes, you can add any mobile number as subscription to an SNS topic.
So we would configure two different SNS topics as two different alert channels in Grafana:
a) 247 <- Subscribe to 247's SQS queue
b) SMS <- Subscribe to the on-call phone via text (if we want to still get text alerts)

**Comment by zya** [comment]
Cool 👍

**Pull request merged**